refactor/ast.py

```diff
@@ -51,6 +52,32 @@ def apply_indentation(
         if len(self.data) >= 1:
             self.data[-1] += str(end_suffix)  # type: ignore
 
+    def apply_indentation_from_source(
+        self,
+        indentation: StringType,
+        source_data: List[StringType],
+        *,
+        start_prefix: AnyStringType = "",
+        end_suffix: AnyStringType = "",
+    ) -> None:
+        """Apply the given indentation only if the corresponding line in the source is different,
+        optionally with start and end prefixes to the bound source lines.
+        """
+
+        def _is_original(i: int) -> bool:
+            common_chars: str = find_common_chars(str(self.data[i]), str(source_data[i].data))
+            is_multiline_string: int = str(self.data[i]).find(common_chars) == 0 and common_chars in ["'''", '"""']
+            return i < len(source_data) and (str(self.data[i]) == common_chars or is_multiline_string)
+
+        for index, line in enumerate(self.data):
+            if index == 0:
+                self.data[index] = indentation + str(start_prefix) + str(line)  # type: ignore
```

Comment:
Is there a case where the first line can still be original?

Reply:
I think it's likely.
From what I understood of the flow, the indentation is inferred from comparing the first line, so it seemed better to not modify this logic
Ok, applied to the first line as well. This PR was buggy, in the case of match to original it did not update self.data[index], this led to the test of `2 + 2 + 3 +4` to fail (the type was `Lines` versus `SourceLines`)